### DISCUSSION TITLE
api  for  signserver/rest/v1/workers/reload

### DISCUSSION THREAD

**Discussion by laimujun** [Keyfactor/signserver-ce#82] [Q&A] [answered]
I use api to activate worker error: “ Admin not authorized to resource. Client certificate authentication required. ”  Could you please tell me the reason for this?

required url:   https://sign.vteamcloud.com/signserver/rest/v1/workers/reload
required parameter: 
{
    "workerIDs":[1]
}

request header:     X-Keyfactor-Requested-With:admin

I don't know what to fill in for the value of X-Keyfactor-Requested-With, so just fill in an admin

**Comment by mlundblad** [comment, marked as answer]
That error message means that the REST operation requires admin privileges.
So, in this case it would require a client-certificate that is configured in SignServer as an authorized admin (or, if "Allow any" option is set, any client-certificate issued by the same CA as the server TLS certificate would be consider an administrator).

The X-Keyfactor-Requested-With header is required as a measure to prevent XSRF: https://en.wikipedia.org/wiki/Cross-site_request_forgery

As long as the header is set, any value will be OK.

**Reply by laimujun** [reply]
Thanks, the issue was resolved after the certificate information was added to send the request

**Reply by r-shafi** [reply]
@laimujun would you kindly let me know of the data you passed?
I am currently passing `subject=C = x, ST = x, L = x, O = x, OU = x, CN = x` but getting the response: `{
  "error": "Admin not authorized to resource. Client certificate authentication required."
}`

**Reply by netmackan** [reply]
@laimujun You need to make sure the request you are sending are using client certificate authentication, i.e. you need to configure the application you are using with access to your private key and certificate.
